charmcraft/services/store.py

```diff
@@ -164,6 +165,16 @@ class StoreService(BaseStoreService):
 
     ClientClass = store.Client
     client: store.Client  # pyright: ignore[reportIncompatibleVariableOverride]
+    anonymous_client: store.AnonymousClient
+
+    @override
+    def setup(self) -> None:
+        """Set up the store service."""
+        super().setup()
+        self.anonymous_client = store.AnonymousClient(
```

Comment:
is this feature (anonymous access) something that we should (eventually) upstream in BaseStoreService?